### ISSUE TITLE
Doesn't work with web app Program Entrypoint

### ISSUE THREAD

**Issue by mehdihadeli** [mehdihadeli/WebApplicationFactoryWithHost-Sample#3]
@mauroservienti `WebAppFactory` doesn't work with Program entrypoint.
- [https://github.com/mehdihadeli/WebApplicationFactoryWithHost-Sample/tree/bug-program](https://github.com/mehdihadeli/WebApplicationFactoryWithHost-Sample/tree/bug-program)

**Comment by mauroservienti** [comment]
No, it doesn't. The web host factory is designed to create a test web host inside the test process. You cannot use it to run an external executable even if it's using a similar hosting model.

What are you trying to achieve?

**Comment by mehdihadeli** [comment]
Ok, So this approach is suitable for test components in an isolation, not a real web application and for `integration` and `end-to-end` testing our web application we should use `WebApplicationFactory<EntryPoint>` with `Program.cs` or `Startup.cs` entry point, correct?
